lib/public/AppFramework/Bootstrap/IRegistrationContext.php

```diff
@@ -130,4 +130,19 @@ public function registerMiddleware(string $class): void;
 	 * @since 20.0.0
 	 */
 	public function registerSearchProvider(string $class): void;
+
+	/**
+	 * Register a lazy loaded initial state provider. This will be called once
+	 * the initial state is actually invoked.
+	 *
+	 * If you only need your initial state when your Controller is loaded. Please
+	 * use the InitialState service!
+	 *
+	 * @param string $class Class must extend OCP\AppFramework\Base\InitialState
+	 *
+	 * @return void
+	 *
+	 * @since 20.0.0
+	 */
+	public function registerInitialState(string $class): void;
```

Comment:
```suggestion
	public function registerInitialStateProvider(string $class): void;
```

because what you register is not the state but a class that provides it